metaserver/src/main/scala/scala/meta/languageserver/ScalametaEnrichments.scala

```diff
@@ -52,6 +52,33 @@ object ScalametaEnrichments {
       // case ??? => SymbolKind.Array
       case _ => SymbolKind.Field
     }
+
+    /** Fully qualified name for packages, normal name for everything else */
+    def qualifiedName: Option[String] = tree match {
+      case Term.Name(name) => Some(name)
+      case Term.Select(qual, name) =>
+        qual.qualifiedName.map { prefix =>
+          s"${prefix}.${name}"
+        }
+      case Pkg(sel: Term.Select, _) => sel.qualifiedName
+      case m: Member => Some(m.name.value)
+      case _ => None
+    }
+
+    /** All names within the node.
+     * - if it's a package, it will have its qualified name: `package foo.bar.buh`
+     * - if it's a val/var, it may contain several names in the pattern: `val (x, y, z) = ...`
```

Comment:
You will actually need additional logic to handle `val (a, b)`, it currently handles only `val a, b` which is different. I think we may produce weird/local symbols for `val (a, b)` because of desugarings in scalac so that can be left to be fixed in the future.

Reply:
Thanks for pointing it out. I'll check it 🤔

Reply:
No need to do it here, I think `val (a, b)` will need special treatment in scalameta similar to other patterns like `val Foo(a, b) = `

Reply:
OK. I'll leave it here as is. But does this simple change solve the problem or is it more complicated?

Reply:
For outline I think that solution is fine 👍

Reply:
Done in cb484756b5b01b25bfcbfdde90df436667cbd6fe